src/server/config.js

```diff
@@ -1,48 +1,47 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 
-var config = {};
+/**
+ * The Configuration model manages the configuration properties for the simulation.
+ * These properties are mapped with the options provided for the simulation, and
+ * are cross to any other instance involved in the simulation.
+ * @constructor
+ */
+function Configuration() {
+    this._config = {};
+}
 
-// module properties
 [
     { name: 'debugHostHandlers', optional: true },
     { name: 'forcePrepare' },
     { name: 'liveReload' },
-    { name: 'platform' },
-    { name: 'platformRoot' },
-    { name: 'projectRoot', single: true },
-    { name: 'server', optional: true },
     { name: 'simHostOptions' },
     { name: 'simulationFilePath' },
     { name: 'telemetry', optional: true },
     { name: 'touchEvents', optional: true },
     { name: 'xhrProxy', optional: true }
 ].forEach(function (prop) {
-    Object.defineProperty(module.exports, prop.name, {
+    Object.defineProperty(Configuration.prototype, prop.name, {
```

Comment:
`Object.definedProperty()` will cause `TypeError` in this form (re-defining setter and getter). This should be wrapped into a `try/catch`, expecting a `TypeError`.